### PR TITLE
Add --yield option to note.add (SYN-6031)

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

File: synapse/tests/test_model_base.py [hunk 1]
```diff
@@ -46,6 +46,10 @@ async def test_model_base_note(self):
             self.ne(nodes[0].ndef, nodes[1].ndef)
             self.eq(nodes[0].get('text'), nodes[1].get('text'))
 
+            nodes = await core.nodes('[ inet:fqdn=vertex.link inet:fqdn=woot.com ] | note.add --yield "yieldnote"')
+            self.len(1, nodes)
+            self.eq(nodes[0].get('text'), 'yieldnote')
```
Comment (vEpiphyte) [inline synapse/tests/test_model_base.py:51]: For good hygiene we should also have note.add tests where there are no inbound nodes which do not end up linked to anything.

Comment (Cisphyx) [reply synapse/tests/test_model_base.py:51]: Should that still add the note?

Comment (Cisphyx) [reply synapse/tests/test_model_base.py:51]: I assume it should, we don't specify anything about requiring inbound nodes.

Comment (Cisphyx) [reply synapse/tests/test_model_base.py:51]: added test

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**